### PR TITLE
Update third_party/upb

### PR DESCRIPTION
**Pull request by AspirinSJL**
To pick up the fix in https://github.com/protocolbuffers/upb/pull/164

We need to update upb internally before this PR can be imported.

### CODE REVIEW

File: bazel/grpc_deps.bzl [hunk 1]
```diff
@@ -202,8 +202,8 @@ def grpc_deps():
     if "upb" not in native.existing_rules():
         http_archive(
             name = "upb",
-            strip_prefix = "upb-ed9faae0993704b033c594b072d65e1bf19207fa",
-            url = "https://github.com/google/upb/archive/ed9faae0993704b033c594b072d65e1bf19207fa.tar.gz",
+            strip_prefix = "upb-f9dab27de3f3a77650199da487dd450a49ed35b2",
```
Comment (jtattermusch) [inline bazel/grpc_deps.bzl]: nit: add sha256  the same way as in https://github.com/grpc/grpc/pull/18718?

Comment (AspirinSJL) [reply bazel/grpc_deps.bzl]: I searched but couldn't find how to check the SHA256 of a github archive.. Do you know how to do that?

Comment (AspirinSJL) [reply bazel/grpc_deps.bzl]: @KeithMoyer Could you tell me where to find the SHA256 hash? Thanks!

Comment (KeithMoyer) [reply bazel/grpc_deps.bzl]: Easiest way:
```
curl -sL https://github.com/google/upb/archive/f9dab27de3f3a77650199da487dd450a49ed35b2.tar.gz | sha256sum
13e796fefd33473ef4b016abae6c0e9bb55195c06638dd897d55af001d9382f6  -
$
```

Comment (AspirinSJL) [reply bazel/grpc_deps.bzl]: Thanks!

**Comment by AspirinSJL** [comment]
@haberman So we can merge and import this PR now, right?

**Comment by AspirinSJL** [comment]
We can move forward when `upb_proto_library()` is available internally.

**Comment by AspirinSJL** [comment]
Stale PR.

**Pull request closed without merging**